gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/AuthController.groovy

```diff
@@ -70,7 +75,11 @@ class AuthController {
    */
   @RequestMapping(value = "/roles/sync", method = RequestMethod.POST)
   void sync() {
-    userRolesSyncer.sync()
+    if (userRolesSyncer) {
```

Comment:
I'd rather keep everything in master and protect it with a feature flag. The changes aren't so severe that it's incomprehensible to have the two side by side, and it makes it significantly easier to a.) develop against master, and b.) have end users eventually test it out without having to hop on dev branches.